solver/src/liquidity.rs

```diff
@@ -140,6 +157,14 @@ impl Settleable for ConstantProductOrder {
     }
 }
 
+impl Settleable for WeightedProductOrder {
+    type Execution = AmmOrderExecution;
```

Comment:
Nice! I thought we would need a separate `Execution` type for weighted pools, but we don't. Pleasant surprise :smile:.